### ISSUE TITLE
SPACE key does not work under Sway/Wayland

### ISSUE THREAD

**Issue by xyproto** [glfw/glfw#1280]
Hi,

OS and version: 64-bit Arch Linux (rolling)
Release or commit: glfw 3.2.1 compiled for wayland (the "glfw-wayland" package on Arch)

GLFW does not seem to detect when SPACE is pressed.

I'm using the Sway window manager (which uses the Wayland protocol) on 64-bit Arch Linux.

Here's a simple test program:

```cpp
#include <GL/glew.h>
#include <GLFW/glfw3.h>
#include <cstdio>

bool running = false;

void error_callback(int error, const char* description)
{
    fprintf(stderr, "Error: %s\n", description);
}

void key_callback(GLFWwindow* window, int key, int scancode, int action, int mods)
{
    switch (key) {
    case GLFW_KEY_ESCAPE:
        if (action == GLFW_PRESS)
            running = false;
        break;
    case GLFW_KEY_X:
        if (action == GLFW_PRESS) {
            printf("%s\n", "x pressed");
        } else if (action == GLFW_RELEASE) {
            printf("%s\n", "x released");
        }
        break;
    case GLFW_KEY_SPACE:
        if (action == GLFW_PRESS) {
            printf("%s\n", "space pressed");
        } else if (action == GLFW_RELEASE) {
            printf("%s\n", "space released");
        }
        break;
    default:
        break;
    }
}

int main(int argc, char* argv[])
{
    const size_t buffer_width = 224;
    const size_t buffer_height = 256;

    glfwSetErrorCallback(error_callback);

    if (!glfwInit())
        return -1;

    glfwWindowHint(GLFW_OPENGL_PROFILE, GLFW_OPENGL_CORE_PROFILE);
    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 3);
    glfwWindowHint(GLFW_OPENGL_FORWARD_COMPAT, GL_TRUE);

    /* Create a windowed mode window and its OpenGL context */
    GLFWwindow* window
        = glfwCreateWindow(2 * buffer_width, 2 * buffer_height, "Space Invaders", nullptr, nullptr);
    if (!window) {
        glfwTerminate();
        return -1;
    }

    glfwSetKeyCallback(window, key_callback);

    glfwMakeContextCurrent(window);

    GLenum err = glewInit();
    if (err != GLEW_OK) {
        fprintf(stderr, "Error initializing GLEW.\n");
        glfwTerminate();
        return -1;
    }

    running = true;
    while (!glfwWindowShouldClose(window) && running) {
        glfwPollEvents();
    }

    glfwDestroyWindow(window);
    glfwTerminate();

    return 0;
}
```

One possible way to build it:

    g++ -o main -std=c++17 -O2 main.cpp -lglfw -lGLEW

When running the program, with the window focused, I get the following output when pressing the `x` key a couple of times:

```
x pressed
x released
x pressed
x released
x pressed
x released
```

When pressing space, there is no such output.

**Comment by linkmauve** [comment]
I can’t reproduce with GLFW master on a recent Sway (nor on Weston), both x and space work properly.

Note that GLFW 3.2 is very old at this point, many improvements have been added to the Wayland backend since then.

**Comment by xyproto** [comment]
Thanks for attempting to reproduce the issue. I will try again with the latest version of Sway.

**Comment by xyproto** [comment]
My keyboard layout is also non-English; it's Norwegian.

**Comment by elmindreda** [comment]
@Svenny The issue on X11 was indeed a bug in GLFW. It has been fixed in the `3.3-stable` branch and will be included in 3.3.3.

The OP Wayland bug is unrelated afaik.

**Comment by xyproto** [comment]
I can no longer reproduce the issue. It works here now.
